modules/core/arrow-annotations-processor/src/main/java/arrow/generic/CoproductFileGenerator.kt

```diff
@@ -0,0 +1,194 @@
+package arrow.generic
+
+import com.squareup.kotlinpoet.AnnotationSpec
+import com.squareup.kotlinpoet.ClassName
+import com.squareup.kotlinpoet.FileSpec
+import com.squareup.kotlinpoet.FunSpec
+import com.squareup.kotlinpoet.KModifier
+import com.squareup.kotlinpoet.LambdaTypeName
+import com.squareup.kotlinpoet.ParameterSpec
+import com.squareup.kotlinpoet.ParameterizedTypeName
+import com.squareup.kotlinpoet.PropertySpec
+import com.squareup.kotlinpoet.TypeSpec
+import com.squareup.kotlinpoet.TypeVariableName
+import java.io.File
+import com.squareup.kotlinpoet.ParameterizedTypeName.Companion.parameterizedBy
+
+private val genericsToClassNames = mapOf(
+        "A" to "First",
+        "B" to "Second",
+        "C" to "Third",
+        "D" to "Fourth",
+        "E" to "Fifth",
+        "F" to "Sixth",
+        "G" to "Seventh",
+        "H" to "Eighth",
+        "I" to "Ninth",
+        "J" to "Tenth",
+        "K" to "Eleventh",
+        "L" to "Twelfth",
+        "M" to "Thirteenth",
+        "N" to "Fourteenth",
+        "O" to "Fifteenth",
+        "P" to "Sixteenth",
+        "Q" to "Seventeenth",
+        "R" to "Eighteenth",
+        "S" to "Nineteenth",
+        "T" to "Twentieth",
+        "U" to "TwentyFirst",
+        "V" to "TwentySecond"
+)
+
+fun generateCoproducts(destination: File) {
+    for (size in 2 until genericsToClassNames.size + 1) {
+        val generics = genericsToClassNames.keys.toList().take(size)
+
+        FileSpec.builder("arrow.generic.coproduct$size", "Coproduct$size")
+                .apply {
+                    addCoproductClassDeclaration(generics)
```

Comment:
We may want to move some of these functions down the road to a common place for other processors to have them available.